### PR TITLE
Improve tests

### LINKED ISSUE #177: Improve wildfly_validation and tests
##### SUMMARY
I have following topic that could be addressed:
- test form [wildfly_validation/yaml_setup.yml](https://github.com/ansible-middleware/wildfly/blob/main/roles/wildfly_validation/tasks/yaml_setup.yml) are valid (IIUC) only for [yaml_config_validation](https://github.com/ansible-middleware/wildfly/tree/main/molecule/yaml_config_validation) tests and should be only there - now every run of wildfly_validation runs tests that may be invalid for the the test.
- add check for HTTPS
- wildfly is in expected directory
- wildfly and all files/dirs are owned by expected user
- wildfly and all files/dirs belong to expected group

##### ISSUE TYPE
 - Enahnce tests

### PR DESCRIPTION
**Pull request by istraka**
Fixes #177

### CODE REVIEW

**Comment by istraka** [comment]
Hi @rpelisse 

I am not sure about the level of support for Windows and if [the shell command](https://github.com/ansible-middleware/wildfly/pull/178/files#diff-accc438fd7e3eec785e0c42ace1303eb351b70b402c9c71e124a92f1fa112373R77) would work correctly in such case?

File: molecule/yaml_config_validation/molecule.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 driver:
-  name: docker
+  name: podman
```
Comment (guidograzioli) [inline molecule/yaml_config_validation/molecule.yml:3]: revert

File: molecule/yaml_config_validation/verify.yml [hunk 2]
```diff
@@ -7,4 +7,15 @@
   collections:
     - middleware_automation.wildfly
   roles:
-    - wildfly_validation
\ No newline at end of file
+    - wildfly_validation
+  tasks:
+    - name: "Verify server's internal configuration"
+      ansible.builtin.include_tasks: ../../playbooks/roles/wildfly_validation/tasks/verify_with_cli_queries.yml
```
[the remark this answers is not in the document]
Comment (istraka) [reply molecule/yaml_config_validation/verify.yml]: Done

File: molecule/yaml_config_validation/verify.yml [hunk 3]
```diff
@@ -7,4 +7,15 @@
   collections:
     - middleware_automation.wildfly
   roles:
-    - wildfly_validation
\ No newline at end of file
+    - wildfly_validation
+  tasks:
```
[the remark this answers is not in the document]
Comment (istraka) [reply molecule/yaml_config_validation/verify.yml]: Done

File: roles/wildfly_validation/tasks/main.yml [hunk 4]
```diff
@@ -46,17 +46,72 @@
...
+        find ./ -printf '%u %p\n' | grep -q -v ^{{ wildfly_user | quote }}
+    chdir: "{{ wildfly_home }}"
+  register: wildfly_files_wrong_user
+
+- name: "Print files or directories owned by wrong user"
+  debug:
+    msg: "{{ wildfly_files_wrong_user }}"
+
+- fail: msg="There are files or directories owned by different user than {{ wildfly_user }}"
+  when: (wildfly_files_wrong_user is not defined) or (wildfly_files_wrong_user|length > 0)
```
[the remark this answers is not in the document]
Comment (istraka) [reply roles/wildfly_validation/tasks/main.yml]: Yeah, basically the output must be empty. I went with `.stdout`.

File: roles/wildfly_validation/tasks/verify_with_cli_queries.yml [hunk 5]
```diff
@@ -43,6 +44,32 @@
...
     fail_msg: "CLI query was not successful: {{ validation_query }}, server may not be functionnal!"
+
+- name: "Check that query result is not undefined."
+  ansible.builtin.assert:
+    that:
+      - query_result.result | type_debug | string != "NoneType"
+    quiet: true
+    fail_msg: "CLI query was not successful: {{ validation_query }}, server may not be functionnal!"
+  when:
+    - expected_result_undefined is not defined
```
[the remark this answers is not in the document]
Comment (istraka) [reply roles/wildfly_validation/tasks/verify_with_cli_queries.yml]: This is a case when the WildFly attribute is undefined. It looks like this:
```
$ ./bin/jboss-cli.sh -c --command="/subsystem=ee/service=default-bindings:read-attribute(name=jms-connection-factory)"
{
    "outcome" => "success",
    "result" => undefined
}
```

using JSON output (`jboss_cli` task):
```
$ ./bin/jboss-cli.sh -c --command="/subsystem=ee/service=default-bindings:read-attribute(name=jms-connection-factory)"  --output-json
{
    "outcome" : "success",
    "result" : null
}
```

Customer may want to verify, that something is not defined.

File: molecule/yaml_config_validation/molecule.yml (hunk 1 above)
Comment (istraka) [reply molecule/yaml_config_validation/molecule.yml]: Done.

**Review by guidograzioli — APPROVED** [review]
LGTM

**Pull request merged**